nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,337 @@
+        } else {
+            final String uri, accessKey;
+            uri =  context.getProperty(URI).getValue();
+            accessKey = context.getProperty(DB_ACCESS_KEY).getValue();
+            final ConsistencyLevel clevel;
+            final String selectedConsistency = context.getProperty(CONSISTENCY).getValue();
+
+            switch(selectedConsistency) {
+                case CONSISTENCY_STRONG:
+                    clevel =  ConsistencyLevel.STRONG;
+                    break;
+                case CONSISTENCY_CONSISTENT_PREFIX:
+                    clevel = ConsistencyLevel.CONSISTENT_PREFIX;
+                    break;
+                case CONSISTENCY_SESSION:
+                    clevel = ConsistencyLevel.SESSION;
+                    break;
+                case CONSISTENCY_BOUNDED_STALENESS:
+                    clevel = ConsistencyLevel.BOUNDED_STALENESS;
+                    break;
+                case CONSISTENCY_EVENTUAL:
+                    clevel = ConsistencyLevel.EVENTUAL;
+                    break;
+                default:
+                    clevel = ConsistencyLevel.SESSION;
+            }
+            if (cosmosClient != null) {
+                closeClient();
+            }
+            if(logger.isDebugEnabled()) {
+                logger.debug("Creating CosmosClient");
+            }
+
+            createCosmosClient(uri, accessKey, clevel);
+        }
+        getCosmosDocumentContainer(context);
+    }
+
+    protected void createCosmosClient(final String uri, final String accessKey, final ConsistencyLevel clevel) throws NullPointerException {
+        try {
+            ConnectionPolicy connectionPolicy = ConnectionPolicy.getDefaultPolicy();
+            this.cosmosClient = new CosmosClientBuilder()
+                .endpoint(uri)
+                .key(accessKey)
+                .connectionPolicy(connectionPolicy)
+                .consistencyLevel(clevel)
+                .buildClient();
+        } catch (NullPointerException e) {
+            getLogger().error("Failed to build cosmosClient due to {}",  e);
+            throw e;
+        }
+    }
+
+    protected abstract void doPostActionOnSchedule(final ProcessContext context);
+
+    protected void getCosmosDocumentContainer(final ProcessContext context) throws CosmosClientException {
+        final String databaseName = context.getProperty(DATABASE_NAME).getValue();
+        final String containerID = context.getProperty(CONTAINER_ID).getValue();
+        final String partitionKey = context.getProperty(PARTITION_KEY).getValue();
+
+        CosmosDatabase database = this.cosmosClient.createDatabaseIfNotExists(databaseName).getDatabase();
+
+        CosmosContainerProperties containerProperties =
+            new CosmosContainerProperties(containerID, "/"+partitionKey);
+
+        //  Create container with 1000 RU/s by default if Not exists
+        this.container = database.createContainerIfNotExists(containerProperties, 1000).getContainer();
```

Comment:
Should this 1000 value be a property?

Reply:
This is the default at the Azure portal, so it seems reasonable to use.

Reply:
I agree with Muazma. The right place to control Cosmos RU is from Azure portal. My code is is intended to create a container if not exists. This RU can be updated thru Azure portal and cosmos control panel at any time, and which will be mis-matched if we provide the property to this NIFI processor.